homeassistant/components/weatherkit/weather.py

```diff
@@ -0,0 +1,255 @@
+    "Frigid": "snowy",
+    "Hail": "hail",
+    "Hot": "sunny",
+    "Flurries": "snowy",
+    "Sleet": "snowy",
+    "Snow": "snowy",
+    "SunFlurries": "snowy",
+    "WintryMix": "snowy",
+    "Blizzard": "snowy",
+    "BlowingSnow": "snowy",
+    "FreezingDrizzle": "snowy-rainy",
+    "FreezingRain": "snowy-rainy",
+    "HeavySnow": "snowy",
+    "Hurricane": "exceptional",
+    "TropicalStorm": "exceptional",
+}
+
+
+def _map_daily_forecast(forecast) -> Forecast:
+    return {
+        "datetime": forecast.get("forecastStart"),
+        "condition": condition_code_to_hass[forecast.get("conditionCode")],
+        "native_temperature": forecast.get("temperatureMax"),
+        "native_templow": forecast.get("temperatureMin"),
+        "native_precipitation": forecast.get("precipitationAmount"),
+        "precipitation_probability": forecast.get("precipitationChance") * 100,
+        "uv_index": forecast.get("maxUvIndex"),
+    }
+
+
+def _map_hourly_forecast(forecast) -> Forecast:
+    return {
+        "datetime": forecast.get("forecastStart"),
+        "condition": condition_code_to_hass[forecast.get("conditionCode")],
+        "native_temperature": forecast.get("temperature"),
+        "native_apparent_temperature": forecast.get("temperatureApparent"),
+        "native_dew_point": forecast.get("temperatureDewPoint"),
+        "native_pressure": forecast.get("pressure"),
+        "native_wind_gust_speed": forecast.get("windGust"),
+        "native_wind_speed": forecast.get("windSpeed"),
+        "wind_bearing": forecast.get("windDirection"),
+        "humidity": forecast.get("humidity") * 100,
+        "native_precipitation": forecast.get("precipitationAmount"),
+        "precipitation_probability": forecast.get("precipitationChance") * 100,
+        "cloud_coverage": forecast.get("cloudCover") * 100,
+        "uv_index": forecast.get("uvIndex"),
+    }
+
+
+class WeatherKitWeather(
+    SingleCoordinatorWeatherEntity[WeatherKitDataUpdateCoordinator]
+):
+    """Weather entity for Apple WeatherKit integration."""
+
+    _attr_attribution = ATTRIBUTION
+
+    _attr_has_entity_name = True
+    _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_native_pressure_unit = UnitOfPressure.MBAR
+    _attr_native_visibility_unit = UnitOfLength.KILOMETERS
+    _attr_native_wind_speed_unit = UnitOfSpeed.KILOMETERS_PER_HOUR
+
+    _attr_supported_features = (
+        WeatherEntityFeature.FORECAST_DAILY | WeatherEntityFeature.FORECAST_HOURLY
+    )
+
+    def __init__(
+        self,
+        coordinator: WeatherKitDataUpdateCoordinator,
+        config: MappingProxyType[str, Any],
+    ) -> None:
+        """Initialise the platform with a data instance and site."""
+        super().__init__(coordinator)
+        self._config = config
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID."""
+        location = self._config[CONF_LOCATION]
+        return f"{location[CONF_LATITUDE]}-{location[CONF_LONGITUDE]}"
```

Comment:
This can be a shorthand attribute in the constructor `_attr_unique_id = ...`

Reply:
I was also wondering if this unique ID should be prefixed with the domain, or if it is unique within the domain. Because I imagine users may have several entities linked to their home location.

Reply:
no domain needed, its already considered

https://github.com/home-assistant/core/blob/d59aa958b6d6eda6d8aef3dc406edce1a7301459/homeassistant/components/homekit/aidmanager.py#L38